_projects/enhance-and-improve-coala-community-and-projects-website.md

```diff
@@ -0,0 +1,160 @@
+to the organization before making any contributions, so this project adds a
+form for newcomers who wish to join the coala community. Also, to give a
+motivation to existing contributors, coala has created a `Promotion form` which
+will promote them to `coala Developer` status. To reduce the workload for
+maintainers, this project includes the development of form with some
+rudimentary checks that will promote the coala-newcomer to coala-developer.
+Moreover, this project adds major enhancements to websites that will help
+newcomers as well as developers to know more about this community and about
+what they have contributed to it.
+
+### Goals
+
+* Add calendar app to show timeline for various events
+* Add a one-stop tab for all google forms
+* Add contributors geolocation
+* Add *Login/SignUp* using GitHub
+* Add organization teams
+* Advanced filters
+* Automate welcoming process for newcomer
+* Automate newcomer promotion
+* Create new database models to manage data
+* Ease navigatibility in websites
+* Improve the **UI/UX of websites** to make more interactive
+* Motivating newcomers & developers with -
+  * modifications in Gamification Leaderboard
+  * adding a tab of **GSoC** in projects website
+* New coala-community logo
+* Contributors profile page
+* Admin/Maintainers dashboard
+
+### Milestones
+
+#### PREPARATION/BONDING
+
+* Detailed research on what **new cron jobs** or **API endpoints** need to be
+  added in [coala webservices](https://gitlab.com/coala/landing/).
+* Discussion on improvements to be made in coala [community website](http://community.coala.io/) and
+  [projects website](http://projects.coala.io/)
+* Research on new models to be created (if any) and adding new fields to
+  existing db models in coala community website **db models**.
+* *Created* and *Merged* a cEP of how the new enhancements will be implemented
+  and how will it be beneficial to [coala community](https://github.com/orgs/coala/teams/coala-newcomers).
+
+#### CODING PHASE 1
+
+1. Implementation of methods in **coala [webservices](https://gitlab.com/coala/landing)**
+2. Implementation of listed enhancements in coala [community website](http://community.coala.io/)
+
+    a. Create new **[Community Logo](https://github.com/coala/community/issues/55) + Header Image of README.md**
+
+    b. Add information about new endpoints at coala-webserivces homepage.
+
+    c. Add [contributor's Geolocation](https://github.com/coala/community/issues/18)
+
+    d. Add [Organisation Teams](https://github.com/coala/community/issues/12)
+
+    e. Add [Mentors](https://github.com/coala/community/issues/11)
+
+    f. Add **Newcomer Invitation** [form](https://github.com/coala/community/issues/89#issuecomment-468955279)
+
+    g. Add **Promotion to Developer** [form](https://github.com/coala/community/issues/89)
+
+    h. Manage live data of community members
+
+    i. Addition of more data fields in **Users** db model
+
+    j. Add **Login and Sign-up** using GitHub
```

Comment:
community project is a static project we are not storing any user data, it would not be feasible to implement this.

also why do we need to have these functionalities?

Reply:
Without Authentication, GitHub API calls limit to only 60 per hour. The main objective to add `login` is to increase the limit to 5k per hour. 
Since we can add a single OAuth token (created by the organization) but that won't help because updating the records in the database for all users will surely increase the 5k limit. Also, there will be other jobs running in the background. 
So, adding `login` will help to update the user records on a regular basis with a limit up to 5k per hour.

Reply:
Can't we make it feasible by not removing `db.sqlite3`, that is [removed ](https://github.com/coala/community/blob/master/.ci/build.sh#L5)whenever CI runs on a new change ?

Reply:
No, community website will not be having any database, we just consume apis, do some analysis on that if needed and show it. Anything you want to implement which required data storage, implement it in the landing project, build apis and use it here.

Reply:
I forgot to tell earlier that for user login, I will be using [django-allauth](https://github.com/pennersr/django-allauth) instead of Django login system. Then, in that case, I guess it will be feasible.

Reply:
>Without Authentication, ...

Why do you want to increase the limit anyway?

We are not hitting github apis directly to get the contributor data. We are using our apis end points, which are in the landing project.

Reply:
Yes I agree that currently we're not hitting the limit, but the [new data fields](https://github.com/coala/projects/pull/719#discussion_r265555277) that I'm looking forward to add will definitely cross the limit if the contributor has done a large number of contributions in the organization.

Reply:
>I forgot to tell earlier that for user login, I will be using django-allauth instead of Django login system. Then, in that case, I guess it will be feasible.

No man, I don't think we want user login, registration or anything like that, this is not a commercial website. We just want to show some data of the contributors who are part of this organization, so that we can do some analysis on that and for that data I think we can use org github api key or something.

see https://gitlab.com/coala/landing/blob/master/backend/org/cron.py on how we are getting the contributors data as of now.

Reply:
Sry, I couldn’t rply late night. Got into bed due to midterm exam having today. Will look the cron file, to get more information and will get back too u soon 🙂

Reply:
I have looked over how currently org GitHub API key is in use. We can also use it to perform new cron jobs that I am looking forward to adding into coala-webservices. But we'll have to take care of jobs being performed because we have an only 5k limit per hour.  So, jobs will be carried out at different times(every hour). 
So, should I remove this from PR as well as from application?

Reply:
But `login` was a way for authenticated users to see the profile from which they will have access to new enhancements -
1. Promotion to developer status
2. Getting assigned to an issue
3. Participate as mentor
4. Extension to non-availability period during GSoC
5. Google forms 
It was a way to avoid spamming. Removal of authentication can lead to spam.
Other users can also check the profile of other users but won't be able to options stated above.

Reply:
>1. Promotion to developer status...

I believe we could do these thing from the leaderboard dashboard right?

Anyway, we couldn't have a separate profile page for each contributor. That would require generating thousands of pages, which would take time, which would hit netlify time limit. 

Also we have another problem currently, we don't know why netlify fail when deploying more number of pages. see https://github.com/coala/community/pull/152, so you could have the same problem while generating separate pages for each contributor.

Reply:
> leaderboard dashboard right?

We can do this by providing that option either beside the username of currently logged in user or by showing it somewhere at the upper-right corner of page/leaderboard.
But I was thinking to add it into the profile.

> we couldn't have a separate profile page for each contributor.
I know that community website is a static website and creating thousands of profile pages for each contributor is not recommendable but can't we make it a little-bit dynamic which can make user-interaction more and spread the word about the community that takes cares of its contributor by helping them to analyze their contributions and their potential. I would be using multi-processing to make API Endpoints call because it has to query nearly 3-4 DB models to display user data on that profile page.
Since the leaderboard page will also be dynamic because stats will change with time.
Making website dynamic is helping a lot to the community to spread its idea to other open-source communities.